stl/inc/xmemory

```diff
@@ -1015,15 +1016,15 @@ struct _Alloc_construct_ptr { // pointer used to help construct 1 _Alloc::value_
 struct _Fake_allocator {};
 
 struct _Container_base0 {
-    void _Orphan_all() noexcept {}
-    void _Swap_proxy_and_iterators(_Container_base0&) noexcept {}
-    void _Alloc_proxy(const _Fake_allocator&) noexcept {}
-    void _Reload_proxy(const _Fake_allocator&, const _Fake_allocator&) noexcept {}
+    _CONSTEXPR20 void _Orphan_all() noexcept {}
+    _CONSTEXPR20 void _Swap_proxy_and_iterators(_Container_base0&) noexcept {}
+    _CONSTEXPR20 void _Alloc_proxy(const _Fake_allocator&) noexcept {}
+    _CONSTEXPR20 void _Reload_proxy(const _Fake_allocator&, const _Fake_allocator&) noexcept {}

```

Comment:
```suggestion
    constexpr void _Orphan_all() noexcept {}
    constexpr void _Swap_proxy_and_iterators(_Container_base0&) noexcept {}
    constexpr void _Alloc_proxy(const _Fake_allocator&) noexcept {}
    constexpr void _Reload_proxy(const _Fake_allocator&, const _Fake_allocator&) noexcept {}
```
All of this can be `constexpr` unconditionally.

Reply:
Actually they cannot. The problem is that we are invoking undefined behavior in the iterator machinery. The problem lies in changing the const `_Mynextiter` member of the `_Container_proxy`

Now the problem here is that we cannot make this mutable unconditionally due to a clang bug which rejects construction of mutable subobjects during constant evaluation.

So marking this constexpr unconditionally is not possible currently